accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -737,10 +750,18 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
         let entry = map.entry(pubkey);
         m.stop();
         let new_entry_zero_lamports = new_entry.is_zero_lamport();
+        let mut other_slot = None;
         let (found_in_mem, already_existed) = match entry {
             Entry::Occupied(occupied) => {
                 // in cache, so merge into cache
                 let (slot, account_info) = new_entry.into();
+
+                let slot_list = occupied.get().slot_list.read().unwrap();
+                if slot_list.len() == 1 {
+                    other_slot = Some(slot_list[0].0);
+                }
+                drop(slot_list);
```

Comment:
what do you mean use `that`? 
And what is used for?

Reply:
I think the effect is the same. 
Here we are collection them out side of the function. If we use relcaim, we need to pass the vec inside and function and fill in there. 
We could optimize to pass `IgnoreReclaim` here. 

Another difference is that if we populate inside the function, we are holding a lock. This may hurts performance.

Reply:
dd7389bfdd4eefafea4444d024f0c477d10cc1a7

Reply:
`insert_new_entry_if_missing_with_lock` is used for both disk and no-disk index generation. 

If we change it, we will need to change both code path. I am trying to limit the change for only no-disk case. 
I don't want to change the code for disk index gen, which we know is good.

Reply:
In `lock_and_update_slot_list` we are holing a lock on the index entry. there are multiple threads trying to insert into the index. Therefore, we don't want to do more work there. so i think it is better to just return and let caller collect the value.

Reply:
reclaims is a different concept than what we need here. 

What reclaim does is to remove any items in the slot_list that has the same *slot* and keep only one. 
All the removed items are put in the reclaim vectors. 

Here we want to record the old item's slot from 'slot_list'. The old slot is different from current slot.

Reply:
yeah. 
moved this to 

https://github.com/anza-xyz/agave/pull/4069